repos/omeka-i18n-module/src/Site/LocalizationListener.php

```diff
@@ -43,28 +45,44 @@ class LocalizationListener
     public function __construct(Settings $settings, SiteSettings $siteSettings, Translator $translator)
     {
         $this->settings = $settings;
-        $this->siteSettings = $settings;
+        $this->siteSettings = $siteSettings;
         $this->translator = $translator;
     }
 
     public function __invoke(MvcEvent $event)
     {
         $route = $event->getRouteMatch();
         $globalLocale = $this->settings->get('locale');
-        $fallbackLocale = $route->getParam('__SITE__', false) ? $this->siteSettings->get('locale') : $globalLocale;
+        $isOnSite = $route->getParam('__SITE__', false);
+        $fallbackLocale = $isOnSite ? $this->siteSettings->get('locale') : $globalLocale;
+        $routerLocale = $route->getParam(self::LOCALE_PARAM);
 
         $sessionManager = SessionContainer::getDefaultManager();
         $session = $sessionManager->getStorage();
 
+        /** @var \Zend\I18n\Translator\Translator $delegateTranslator */
         $delegateTranslator = $this->translator->getDelegatedTranslator();
         $locale = $route->getParam(self::LOCALE_PARAM) ?? $session['locale'] ?? $fallbackLocale;
 
         if (null === $locale || empty($locale)) {
             $locale = 'en';
         }
 
+        $isMultiLingual = $isOnSite ? boolval($this->siteSettings->get('i18n-multi-lingual-site')) : false;
+        /** @var TreeRouteStack $router */
+        $router = $event->getRouter();
+
+        if ($isMultiLingual && $routerLocale) {
```

Comment:
Is setting it to null required if it's not present? If so we might as well capture this in a ternary: `$uriLocale = ($isMultilingual && $routerLocale) ? $locale : null;` 

Also, a nitpick: multilingual is one word, so shouldn't be uppercased at Lingual.